internal/admission/handler_test.go

```diff
@@ -0,0 +1,179 @@
+package admission
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	sensor "github.com/argoproj/argo-events/pkg/apis/sensor/v1alpha1"
+	"github.com/kanopy-platform/argoslower/pkg/ratelimit"
+	"github.com/stretchr/testify/assert"
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type fakeRateLimitGetter struct {
+	Rates map[string]*sensor.RateLimit
+	Err   error
+}
+
+func (frlg *fakeRateLimitGetter) RateLimit(namespace string) (*sensor.RateLimit, error) {
+	if r, ok := frlg.Rates[namespace]; ok {
+		return r, nil
+	} else {
+		return nil, frlg.Err
+	}
+}
+
+func newFakeRate() fakeRateLimitGetter {
+	return fakeRateLimitGetter{
+		Rates: map[string]*sensor.RateLimit{},
+	}
+}
+
+func TestSensorMutationHook(t *testing.T) {
+
+	t.Parallel()
+	frlg := newFakeRate()
+
+	testRate := &sensor.RateLimit{
+		Unit:            "Second",
+		RequestsPerUnit: int32(2),
+	}
+
+	frlg.Rates["test"] = testRate
+	frlg.Rates["novalue"] = nil
+
+	rc := ratelimit.NewRateLimitCalculatorOrDie("Second", int32(1))
+	defaultRate := rc.Calculate(nil, nil)
+
+	h := NewHandler(&frlg, rc)
+
+	scheme := runtime.NewScheme()
+	utilruntime.Must(sensor.AddToScheme(scheme))
+	decoder, err := admission.NewDecoder(scheme)
+	assert.NoError(t, err)
+
+	err = h.InjectDecoder(decoder)
+	assert.NoError(t, err)
+
+	sen := sensor.Sensor{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "test",
+		},
+		Spec: sensor.SensorSpec{},
+	}
+
+	tests := []struct {
+		description string
+		trigger     sensor.Trigger
+		ns          string
+		expected    float64
```

Comment:
Is this the expected number of mutations?

Reply:
renamed the field to make it clearer.